### PR TITLE
ci: implement build matrix for JDK

### PR DESCRIPTION
**Pull request by Emrehzl94**
This PR modifies the TeamCity build pipeline by adding build steps specifically for Java 17. Additionally, several dependencies have been updated to ensure compatibility of the Java 17 build steps.

### CODE REVIEW

**Comment by Emrehzl94** [comment]
[Build with Java 17 & 21 as well](https://trello.com/c/u2pddlIs/413-build-with-java-17-21-as-well)

File: .teamcity/builds/Build.kt [hunk 1]
```diff
@@ -15,33 +15,47 @@ class Build(
...
+                    "package",
+                    javaVersion,
+                    "-pl :packaging -am -DskipTests")
+
+            sequential {
+              dependentBuildType(Maven("${name}-build", "build", "test-compile", javaVersion))
+              dependentBuildType(Maven("${name}-unit-tests", "unit tests", "test", javaVersion))
+              dependentBuildType(collectArtifacts(packaging))
+              dependentBuildType(
+                  IntegrationTests("${name}-integration-tests", "integration tests", javaVersion) {
```
Comment (ali-ince) [inline .teamcity/builds/Build.kt:40]: I think one problem with the new setup is that the integration tests run on the desired java version, but we do not do anything about the java version on the Kafka Connect container - which is the actual thing, or am I missing something?

Comment (ali-ince) [reply .teamcity/builds/Build.kt:40]: Hmm, I see now that we also migrated the images to the latest version of Confluent Platform.

**Review by ali-ince — COMMENTED** [review]
I'm not really sure how we should ideally structure this with different versions to build and a matching runtime environment. Let's discuss this with @fbiville when he is back.

File: .teamcity/builds/Common.kt [hunk 2]
```diff
@@ -84,15 +87,20 @@ fun collectArtifacts(buildType: BuildType): BuildType {
...
         localRepoScope = MavenBuildStep.RepositoryScope.MAVEN_DEFAULT
 
         dockerImagePlatform = MavenBuildStep.ImagePlatform.Linux
-        dockerImage = "eclipse-temurin:11-jdk"
+        dockerImage =
+            when (javaVersion) {
+              "11" -> "eclipse-temurin:11-jdk"
+              "17" -> "eclipse-temurin:17-jdk"
+              else -> error("Unsupported Java version: $javaVersion")
+            }
```
Comment (ali-ince) [inline .teamcity/builds/Common.kt:103]: we could define a JavaVersion enum, and derive the image from that and also get rid of this and other similar blocks?

File: pom.xml [hunk 3]
```diff
@@ -82,7 +82,7 @@
         <neo4j-java-driver.version>4.4.18</neo4j-java-driver.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <!-- to match what is shipped with the Kafka Connect version used in our integration test environment -->
-        <protobuf.version>3.19.6</protobuf.version>
+        <protobuf.version>3.24.4</protobuf.version>
```
Comment (ali-ince) [inline pom.xml:85]: why do we need the upgrade?

**Review by ali-ince — CHANGES_REQUESTED** [review]

File: .teamcity/builds/Common.kt (hunk 2 above)
Comment (Emrehzl94) [reply .teamcity/builds/Common.kt:103]: yeah, sounds good 👍

File: pom.xml (hunk 3 above)
Comment (Emrehzl94) [reply pom.xml:85]: version 3.19.6 is not compatible with kafka connect version 7.7.0 but no problems with protobuf 3.24.4 to kafka connect 7.2.9. Another approach would be to change protobuf version according to kafka connect version in integration tests.Wdyt?

Comment (ali-ince) [reply pom.xml:85]: It's not a big problem as this dependency is used during integration tests only.

**Review by ali-ince — APPROVED** [review]

**Pull request merged**